### DISCUSSION TITLE
Detecting Maven releases from Artifactory that have an artifact property set

### DISCUSSION THREAD

**Discussion by hoshposh** [renovatebot/renovate#31026] [Suggest an Idea]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Gitlab

## Overview

While a project is in development using a `-SNAPSHOT` version suffix, our build scripts generate Maven release candidate builds for a project each night as a daily build.

When we finally do release, we create a Github release with the following information:

- Tag: Matches the version used throughout the development, with a `v` prefix and without the `-SNAPSHOT` suffix.
- In the release description the final Maven release version is recorded.
- In Artifactory the release artifact is marked with a property `git.tag`.
- Eventually prune out the release candidate builds from the Maven repository (does not happen straight away, usually at some point after the release) that do not have the property `git.tag`.

Example:

- Version during development is `1.2.0-SNAPSHOT`, which results in daily build versions of `1.2.0-1`, `1.2.0-2`, etc.,.
- Version at final release is `1.2.0-56` which is recorded in the release description.
- Release is tagged `v1.2.0`.
- Artifactory release artifact is updated with a property `git.tag` set to `v1.2.0`.

## Question

How would we alter renovate to look for new releases in the Artifactory repository and only use artifacts that have the property `git.tag` set?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by hoshposh** [comment]
I have dug through other discussions and I think the approach would be as follows:
- Create a [custom datasource](https://docs.renovatebot.com/modules/datasource/custom/) that accesses the Artifactory (JFrog) API endpoints.
- Find an endpoint that will retrieve artifacts for specific GAV attributes but allow filtering based on item properties.

Is it possible to perform `POST` actions with a custom datasource since the [Artifactory Query language](https://jfrog.com/help/r/jfrog-rest-apis/aql-execution) appears to provide the searching capabilities I need?

**Reply by astellingwerf** [reply]
The current JSONFetcher only supports GET requests, but I'd say that adding support different methods makes a lot of sense.

We'd need to figure out whether/how we should use Host Rules for authentication on it too then.

**Reply by astellingwerf** [reply]
We need to decide whether JSONata is the best/appropriate/only engine for generating the (in this case POST) request body. In your example of JFrog AQL, the request body is not JSON, but `text/plain` proprietary format instead.
